lib/text/listener.js

```diff
@@ -83,6 +83,82 @@ class Importer extends SHRParserListener {
     }
   }
 
+    enterValuesetDefsDoc(ctx) {
+        const ns = ctx.valuesetDefsHeader().namespace().getText();
+        this._currentNs = ns;
+        if (typeof this._nsMap[ns] == 'undefined') {
+            this._nsMap[ns] = new Namespace(ns);
+        }
+    }
+
+    exitValuesetDefsDoc(ctx) {
+        // clear current namespace and uses namespaces
+        this._currentNs = '';
+        this._usesNs = [];
+    }
+
+    enterValuesetDef(ctx) {
+        const name = ctx.valuesetHeader().simpleName().getText();
+        const id = new Identifier(this._currentNs, name);
+        const valueset = new Valueset(id);
+
+        var path = this._preprocessedData.resolvePath('default', this._currentNs);
+
+        valueset.url = path.url + name;
+        valueset.name = name;
+
+        this._currentDef = valueset;
+    }
+
+    exitValuesetDef(ctx) {
+      this.pushCurrentDefinition();
+    }
+
+    enterValuesetValue(ctx) {
+        const code = ctx.getText();
+        var value;
+        var result;
+        if (ctx.valuesetInlineValue() != null) {
+            result = this.processInlineValue(ctx.valuesetInlineValue());
+        } else if (ctx.fullyQualifiedCode() != null){   // e.g. MTH#1234567
+            result = this.processFullyQualifiedCode(ctx.fullyQualifiedCode());
```

Comment:
Unfortunately, `this.processFullyQualifiedCode` uses `this._preprocessedData` to resolve the vocabulary, but you really want to use `this._preprocessedVSData` (once you introduce it to this class).  Ugh.  Maybe you want a flag in the importer that is something like `_isValueSetDoc` that gets set in `enterValueSetDefsDoc(ctx)` and unset in `exitValueSetDefsDoc(ctx)`.  And then a function like `getPreprocessedData()` that will return `this._preprocessedVSData` or `this._preprocessedData` depending on if that flag is set or not.  Then functions who need the data (like `processFullyQualifiedCode`) can _always_ use the `this.getPreprocessedData()` function to ensure they get the right one.

Are we having fun yet?